### PR TITLE
[11877] regenerate certificate to work with latest service_identity

### LINKED ISSUE #11877: trunk test failure: twisted.internet.test.test_endpoints.WrapClientTLSParserTests.test_tls fails with service_identity 23.1
tests are failing on trunk again due to service_identity 23.1 release no longer honoring the CN fallback, which our test server.pem uses because it includes no SANs

Updated for posterity; it's this traceback: https://github.com/twisted/twisted/actions/runs/5222913768/jobs/9532027215

```
Traceback (most recent call last):
  File "/home/runner/work/twisted/twisted/.tox/alldeps-withcov-posix/lib/python3.10/site-packages/twisted/internet/test/test_endpoints.py", line 4276, in test_tls
    self.assertFalse(plainClient.transport.disconnecting)
  File "/home/runner/work/twisted/twisted/.tox/alldeps-withcov-posix/lib/python3.10/site-packages/twisted/trial/_synctest.py", line 386, in assertFalse
    super().assertFalse(condition, msg)
  File "/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/unittest/case.py", line 681, in assertFalse
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: True is not false

twisted.internet.test.test_endpoints.WrapClientTLSParserTests.test_tls
```

### PR DESCRIPTION
**Pull request by glyph**
also rewrite all the logic using cryptography rather than pyopenssl

## Scope and purpose

Fixes #11877

### CODE REVIEW

**Comment by glyph** [comment]
review please

**Comment by glyph** [comment]
please review

**Comment by glyph** [comment]
@adiroiban is @chevah-robot broken?

File: src/twisted/test/server.pem [hunk 1]
```diff
@@ -1,97 +1,123 @@
...
-cert.set_serial_number(datetime.now().toordinal())
-cert.gmtime_adj_notBefore(0)
-cert.gmtime_adj_notAfter(60 * 60 * 24 * 365 * 100)
+certificate_bytes = (
+    CertificateBuilder()
+    .serial_number(random_serial_number())
+    .not_valid_before(datetime.now())
+    .not_valid_after(datetime.now() + timedelta(seconds=60 * 60 * 24 * 365 * 100))
+    .subject_name(me)
+    .add_extension(SubjectAlternativeName([DNSName("localhost")]), False)
```
Comment (alex) [inline src/twisted/test/server.pem]: Consider using kwarg `critical=False` for readability

**Review by alex — APPROVED** [review]
one suggested comment, lgtm assuming tests pass

**Comment by adiroiban** [comment]
needs-review

**Pull request merged**